src/RA_Core.cpp

```diff
@@ -56,10 +56,23 @@ inline static unsigned int g_nProcessTimer{ 0U };
 
 BOOL APIENTRY DllMain(HMODULE hModule, DWORD dwReason, _UNUSED LPVOID)
 {
-    if (dwReason == DLL_PROCESS_ATTACH)
-        g_hThisDLLInst = hModule;
+    // exceptions can happen here, they must not be allowed to leave
+    try
+    {
+        if (dwReason == DLL_PROCESS_ATTACH)
+        {
+            // resetting it more than once will cause a deadlock or access violation
+            // Library will be freed when the app ends
+            if (!g_hThisDLLInst)
+                g_hThisDLLInst.reset(hModule);
+        }
 
-    ra::services::Initialization::RegisterCoreServices();
+        ra::services::Initialization::RegisterCoreServices();
+    }
+    catch (const std::exception& e)
+    {
+        RA_LOG_ERR("Exception: %s", e.what());
```

Comment:
I put this in here as a safe-guard as the entry-point of a DLL should not exit via exception.